### ISSUE TITLE
.accessibilityHeadingLevel and VoiceOver support

### ISSUE THREAD

**Issue by MarcSolo** [cvs-health/ios-swiftui-accessibility-techniques#5]
Earlier today, I came across a question about heading level support on iOS in the #native-mobile channel on [web-a11y.slack.com](http://web-a11y.slack.com/).

In the thread, "Genevieve" says

> I just tested adding iOS heading levels in swift app and VO announced them. Fantastic! I didn't know we had this available

If this heading level VoiceOver support can be reproduced, I propose the following updates:

- Update Headings markdown documentation to reflect that .accessibilityHeadingLevel is now supported by VoiceOver
- Update Headings technique to include good/bad example of implementing heading levels

**Comment by pauljadam** [comment]
@MarcSolo I can't reproduce support for heading levels in SwiftUI on the latest version of iOS on my iPhone. I don't hear any heading level or heading trait when using this code .accessibilityHeading(.h1)`. Is there a screenshot, code snippet, or working example where I can reproduce their results?

**Comment by MarcSolo** [comment]
No, sorry I don't have any concrete evidence. Just anecdotal evidence based on comments of others in the accessibility community. One of my colleagues indicated that they are now seeing heading levels in Apple's iOS Messages app. So, this app may the best working example I'm aware of.

**Comment by pauljadam** [comment]
@MarcSolo Thanks for bringing this to my attention and sending me the a11y slack invite! :) I've now determined that the chat room members were using both the heading trait and the heading level property `.accessibilityAddTraits(.isHeader).accessibilityHeading(.h1)` and that's the only way it works. This is even counter to Apple's examples that only use `.accessibilityHeading(.h1)` without the extra header trait. But Apple's code examples do not work. Maybe this is a bug and one day we won't need the extra header trait with the heading level property but for now looks like that's the only way it works. Sadly I also found this out 

* You can't make the Page Title heading have a heading level. For example using `.navigationBarTitle("Headings").accessibilityAddTraits(.isHeader).accessibilityHeading(.h1)` will not make the page title a Heading Level 1.

So if I can't have an H1 heading as the main page title heading level then what's the point of having H2 headings below a heading that is not an H1? Apple is not making this easy as usual!

**Comment by MayurPatel0** [comment]
I think the reason for providing the H2 headings is for sheets, popovers, and overlays.
Logically, the content on these elements is part of the parent page.
As such, there by H2 - H6 headings make sense.

Note: This is just an informative thought, but this is still pointless, as content on the parent page cannot have heading level 1.

**Comment by pauljadam** [comment]
suggested updates completed and closing issue
